### PR TITLE
soc: st: stm32: stm32wbax: Update STM32WBA BLE/HAL according Cube FW 1.3.1

### PR DESCRIPTION
**Pull request by asm5878**
STM32WBA has been updated to include STM32Cube FW 1.3.1:

- BLE HCI driver
- BLE HCI interface
- HAL/LL interfaces

### CODE REVIEW

**Review by erwango — CHANGES_REQUESTED** [review]
Please fix compliance issues reported below.
This will include providing commit message, which will be useful to explain your changes.

**Comment by ajarmouni-st** [comment]
CI compliance for a set of commits can be checked locally using `/scripts/ci/check_compliance.py --commits HEAD~n..HEAD`

**Comment by erwango** [comment]
Please fetch latest main top and rebase.

File: soc/st/stm32/stm32wbax/hci_if/bleplat.c [hunk 1]
```diff
@@ -80,25 +80,6 @@ void MX_RAMCFG_Init(void)
...
-	memset((void *)dst, (int)c, (size_t)n);
-
-	return dst;
-}
-
-int8_t ble_memcmp(const void *a, const void *b, uint16_t n)
-{
-	return (int8_t)memcmp(a, b, (size_t)n);
-}
-
```
Comment (erwango) [inline soc/st/stm32/stm32wbax/hci_if/bleplat.c:101]: Question:
Does it mean that controller blob doesn't requires these anymore from Zephyr ?

File: drivers/bluetooth/hci/hci_stm32wba.c [hunk 2]
```diff
@@ -39,9 +39,16 @@ static K_SEM_DEFINE(hci_sem, 1, 1);
 #define BLE_DYN_ALLOC_SIZE \
 	(BLE_TOTAL_BUFFER_SIZE(CFG_BLE_NUM_LINK, MBLOCK_COUNT))
 
+/* GATT buffer size (in bytes)*/
+#define BLE_GATT_BUF_SIZE \
+		BLE_TOTAL_BUFFER_SIZE_GATT(CFG_BLE_NUM_GATT_ATTRIBUTES, \
+					CFG_BLE_NUM_GATT_SERVICES, \
+					CFG_BLE_ATT_VALUE_ARRAY_SIZE)
```
Comment (erwango) [inline drivers/bluetooth/hci/hci_stm32wba.c:46]: Please fix indentation

**Review by erwango — CHANGES_REQUESTED** [review]

File: soc/st/stm32/stm32wbax/hci_if/bleplat.c (hunk 1 above)
Comment (asm5878) [reply soc/st/stm32/stm32wbax/hci_if/bleplat.c:101]: Yes we don't need anymore these functions

**Review by erwango — APPROVED** [review]

**Review by gautierg-st — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Pull request merged**